src/sitemap-pipe.js

```diff
@@ -20,6 +21,48 @@ import setCustomResponseHeaders from './steps/set-custom-response-headers.js';
 import { PipelineStatusError } from './PipelineStatusError.js';
 import { PipelineResponse } from './PipelineResponse.js';
 
+async function generateSitemap(state) {
+  const {
+    owner, repo, ref, contentBusId, partition, s3Loader, log,
+    previewHost, config: { host: prodCDN } = {},
+  } = state;
+  const ret = await s3Loader.getObject('helix-content-bus', `${contentBusId}/live/sitemap.json`);
+  if (ret.status !== 200) {
+    return ret;
+  }
+  let config;
+  try {
+    config = JSON.parse(ret.body);
+  } catch (e) {
+    log.info('failed to parse /sitemap.json', e);
+    throw new PipelineStatusError(404, `Failed to parse /sitemap.json: ${e.message}`);
+  }
+  const { data } = config;
+  if (!data || !Array.isArray(data)) {
+    throw new PipelineStatusError(404, 'Expected \'data\' array not found in /sitemap.json');
+  }
+  const host = partition === 'preview'
+    ? (previewHost || `${ref}--${repo}--${owner}.hlx.page`)
+    : (prodCDN || `${ref}--${repo}--${owner}.hlx.live`);
```

Comment:
```suggestion
    : (prodCDN || liveHost || `${ref}--${repo}--${owner}.hlx.live`);
```

also use `state.liveHost` if available.